### ISSUE TITLE
How to pass SSH flag -i for id_rsa

### ISSUE THREAD

**Issue by chetankatke** [codercom/sshcode#36]
🤔

**Comment by gufastian** [comment]
Same for port.
Can't get it work :(

**Comment by kylejeske** [comment]
Take a look at [Line31](https://github.com/codercom/sshcode/blob/302943d94ad6c12310746c23ef1c3a8e6f56e97a/main.go#L31)

and at [Line111](https://github.com/codercom/sshcode/blob/302943d94ad6c12310746c23ef1c3a8e6f56e97a/main.go#L111)

```
sshCmdStr = fmt.Sprintf(
"ssh -tt -q -L %v %v %v 'cd %v; %v --host 127.0.0.1 --allow-http --no-auth --port=%v'", 
localPort + ":localhost:" + localPort, *sshFlags, host, dir, codeServerPath, localPort,
)

sshCmd = exec.Command("sh", "-c", sshCmdStr)
```

The whole thing just wraps SSH on the command line .. so based on the code above. The second %v allows for SSH parameters.

Using that .. it could look something like ..
[Line 36](https://github.com/codercom/sshcode/blob/302943d94ad6c12310746c23ef1c3a8e6f56e97a/main.go#L36)
```
Usage: [-skipsync] %v HOST [DIR] [SSH ARGS...]
sshcode 10.0.1.10 ~/mydirectory '-i ~/.ssh/id_rsa'
```

**Comment by gufastian** [comment]
I tried passing parameters with quotes and without quotes, but the result is the same. Script tries to connect only to port 22.

`MacBook:Downloads $ ./sshcode-darwin-amd64 xxx@xxx.xxx.com ~/projects '-p 1234'
2019-04-22 19:28:25 INFO	ensuring code-server is updated...
ssh: connect to host xxx.xxx.com **port 22**: Operation timed out
2019-04-22 19:29:41 FATAL	failed to update code-server: exit status 255`

**Comment by SscSPs** [comment]
Currently, I don't think they are supported, but you can do all that by using ~/.ssh/config file to save your SSH targets

**Comment by ammario** [comment]
Sorry, the command's help is currently out-of-date with the flags. Going to fix that today in #37 

`sshcode --ssh-flags "-p 1234" xxx@xxxx.com` should work

**Comment by ammario** [comment]
Similarly, `sshcode --ssh-flags '-i .ssh/id_rsa' xxx@yyyy.com` should work.

Feel free to re-open this issue if that's still not working for you.
